packages/salesforcedx-vscode-soql/src/editor/soqlEditorProvider.ts

```diff
@@ -40,35 +45,32 @@ export class SOQLEditorProvider implements vscode.CustomTextEditorProvider {
         )
       ]
     };
-
     webviewPanel.webview.html = this.getWebViewContent(webviewPanel.webview);
-  }
+    const instance = new SOQLEditorInstance(document, webviewPanel, _token);
+    this.instances.push(instance);
+    instance.onDispose(this.disposeInstance.bind(this));
 
+    Array.prototype.push.apply(
+      this.context.subscriptions,
+      instance.subscriptions
+    );
+  }
```

Comment:
Its not clear to me why we need this? Why are we calling the prototype too?

Reply:
It's for pushing everything in that array.
I'd recommend this tho for easier readability: `this.context.subscriptions.push(...instance.subscriptions)`

Reply:
Yeah, I assumed we are pushing into an array. I didn't understand why we need to call `apply` on the Array prototype. Why can't we use a simple `push()` on an array. I guess I wanted to know if there is some weirdness going on that required us to use `Array.prototype.push.apply`?

Reply:
simple push pushes the array as an item, while `Array.prototype.push.apply` pushes each individual item

Reply:
Oh yeah, cool, using the ...rest params feature.